File: Algorithms/Sorting/Bubble Sort/bubbleSort.js
const arr = [9, 3, 8, 5, 6, 7, 2, 1, 4];

function bubbleSort(arr) {
  // Start Iterating from the begging till the end.
  for (let i = 0; i <= arr.length; i++) {
    // Start Iterating from the second element upto the element before the last one.
    for (let j = i + 1; j <= arr.length - 1; j++) {
      // Getting sure the order is right.
      if (arr[i] > arr[j]) [arr[i], arr[j]] = [arr[j], arr[i]]; // Swapping if true.
    }
  }
}

bubbleSort(arr);
console.log(arr);

File: Algorithms/Sorting/Bubble Sort/bubble_sort.py
arr = [9, 3, 8, 5, 6, 7, 2, 1, 4]


def bubble_sort(arr):
    for i in range(0, len(arr)):
        for j in range(i + 1, len(arr)):
            if arr[i] > arr[j]:
                [arr[i], arr[j]] = [arr[j], arr[i]]


bubble_sort(arr)
print(arr)
